runtime/test/planner-tests.js

```diff
@@ -621,6 +621,42 @@ describe('Automatic handle resolution', function() {
         B`);
   });
 
+  it('coalesces recipes to resovle connections', async () => {
+    let recipe = await verifyResolvedPlan(`
+      schema Thing
+        Text id
+      schema Product extends Thing
+        Text name
+      schema Other
+        Number count
+      schema Location
+        Number lat
+        Number lng
+
+      particle A
+        out Product product
+      particle B
+        in Thing thing
+        out Other other
+      particle C
+        in * {Number count} something
+        in Location location
+      particle D
+        inout Location location
+
+      recipe
+        A
+      recipe
+        B
+      recipe
+        C
+      recipe
+        D`);
+    assert.lengthOf(recipe.handles, 3);
```

Comment:
You mean asserting on:

"recipe
  create as view0 // ~a
  create as view1 // ~a
  create as view2 // Location {Number lat, Number lng}
  A as particle0
    product -> view0
  B as particle1
    other -> view1
    thing <- view0
  C as particle2
    location <- view2
    something <- view1
  D as particle3
    location = view2"
? 

I feel this is a bad idea. If we start doing that, any change to recipe formatting will cause a need to change many tests in subtle ways. Or did you mean something else?

Reply:
Yes that's what I meant. I don't think it's a big problem having to update test expectations when we change formatting. That will happen less and less over time.

Reply:
My perception is that we risk asserting on too many properties in such case. Tests should be lenient regarding pretty much everything that's not the core thing under test. If we introduce enough tests of this sort it will make it hard to make meaningful changes, as they will require larger than necessary changes across the codebase - I would prefer to avoid that.

Reply:
Changed to what you proposed.